acquire-core-libs/src/acquire-device-properties/device/props/storage.c

```diff
@@ -82,30 +82,120 @@ storage_properties_set_external_metadata(struct StorageProperties* out,
 }
 
 int
-storage_properties_set_chunking_props(struct StorageProperties* out,
-                                      uint32_t chunk_width,
-                                      uint32_t chunk_height,
-                                      uint32_t chunk_planes)
+dimension_init(struct Dimension* out,
+               const char* name,
+               size_t bytes_of_name,
+               enum DimensionType kind,
+               uint32_t array_size_px,
+               uint32_t chunk_size_px,
+               uint32_t shard_size_chunks)
 {
     CHECK(out);
-    out->chunk_dims_px.width = chunk_width;
-    out->chunk_dims_px.height = chunk_height;
-    out->chunk_dims_px.planes = chunk_planes;
+
+    EXPECT(name, "Dimension name cannot be null.");
+    EXPECT(bytes_of_name > 0, "Bytes of name must be positive.");
+    EXPECT(strlen(name) > 0, "Dimension name cannot be empty.");
+    EXPECT(kind > DimensionType_None && kind < DimensionTypeCount,

```

Comment:
This prohibits explicitly using `DimensionType_None` with `dimension_init`, which was surprising to me. Let's say I have a named dimension that doesn't really have a type/kind - should I init it with `Channel`?

Reply:
Getting rid of `DimensionType_None`.